mlir/include/mlir/Analysis/Presburger/Barvinok.h

```diff
@@ -0,0 +1,227 @@
+//===- Barvinok.h - Barvinok's Algorithm -----------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
```

Comment:
Please don't write the whole implementation in the header. Move it into a .cpp file. If the generating function is only used internally, you can move the whole class into the .cpp if you want, no reason to expose it.

Reply:
Do you think it should be the Barvinok.cpp file or a different one?